go/common/node_type.go

```diff
@@ -39,6 +42,22 @@ func (n *NodeType) UnmarshalText(text []byte) error {
 	return nil
 }
 
+func (n NodeType) MarshalJSON() ([]byte, error) {
```

Comment:
It's a bit odd that one of the receivers is a pointer and the other is not

Reply:
Yeah I thought the same but I checked at the time and it is idiomatic in Go. Like I saw [this Jetbrains issue](https://youtrack.jetbrains.com/issue/GO-13587/Add-Marshaler-Unmarshaler-to-exclusions-for-Mixed-value-and-pointer-receivers) saying that the IDE shouldn't flag it as an issue for marshal/unmarshal methods because mixing receiver types for that is what the core libraries do haha. Go's a bit of a mess isn't it 😅